docs/development/local-plugin-creation.md

```diff
@@ -0,0 +1,292 @@
+# Local plugin development guide
+
+So you wanna build a BitOps plugin, eh?  Follow along for how to do it locally!
+
+## 1. Create a plugin repo
+```
+cd /path/to/bitops-plugins
+mkdir my-plugin
+cd my-plugin
+```
+More information on what goes in a plugin [here](../plugins.md).
+
+> Note: If the plugin needs to install tools, you will need to build and run a local version of BitOps.  The first portion of this guide assumes the install.sh script **DOES NOT** needs to be run.  For more information about how to set up BitOps for local development with a plugin that requires an installation, skip to step 5 below.
+
+
+For this example, we'll keep to a really simple plugin:
+`bitops.schema.yaml`
+```
+duplicate-environment:
+    type: object
+    properties:
+      options:
+        type: object
+        properties:
+          foo:
+            type: string
+            export_env: DUPLICATE_ENVIRONMENT_FOO
+            required: True
+            default: bar
+```
+
+`deploy.sh`
+```
+#!/bin/bash
+set -ex
+
+echo "Running Duplicate Environment Plugin deployment script..."
+
+# vars
+export DUPLICATE_ENVIRONMENT_ROOT_SCRIPTS="$BITOPS_PLUGIN_DIR"
+export DUPLICATE_ENVIRONMENT_ROOT_OPERATIONS="$BITOPS_OPSREPO_ENVIRONMENT_DIR"
+export BITOPS_SCHEMA_ENV_FILE="$DUPLICATE_ENVIRONMENT_ROOT_OPERATIONS/ENV_FILE"
+export SCRIPTS_DIR="$DUPLICATE_ENVIRONMENT_ROOT_SCRIPTS/scripts"
+
+if [ ! -d "$DUPLICATE_ENVIRONMENT_ROOT_OPERATIONS" ]; then
+  echo "No duplicate-environment directory.  Skipping."
+  exit 0
+fi
+
+printf "Deploying duplicate-environment..."
+
+if [ ! -f "$BITOPS_SCHEMA_ENV_FILE" ]; then 
+  echo "No duplicate-environment ENV file found"
+else
+  source "$BITOPS_SCHEMA_ENV_FILE"
+fi
+
+bash $SCRIPTS_DIR/validate_env.sh
+
+cd $DUPLICATE_ENVIRONMENT_ROOT_OPERATIONS
+
+echo "Listing contents of duplicate-environment Root: $DUPLICATE_ENVIRONMENT_ROOT_OPERATIONS"
+ls -al .
+
+echo "settings:"
+echo "DUPLICATE_ENVIRONMENT_FOO"
+echo "$DUPLICATE_ENVIRONMENT_FOO"
+```
+> **Note:** Much of the above is best practice boilerplate and is not strictly necessary.
+
+`scripts/validate_env.sh`
+```
+#!/bin/bash
+set -e 
+
+if [ -z "$BITOPS_ENVIRONMENT" ]; then
+  echo "environment variable (ENVIRONMENT) not set"
+  exit 1
+fi
+```
+
+`plugin.config.yaml`
+```
+plugin:
+  deployment:
+    language: bash
+    deployment_script: deploy.sh
+    core_schema_parsing: true
+    life_cycle_scripts: true
+```
+
+
+## 3. Create an ops repo for testing
```

Comment:
Numbers go 1,3,4 

I would suggest breaking these up into their own files under the major header "plugin local development"

Reply:
Updated the numbers for 2 and beyond.

I'll create a ticket for breaking up the docs into their own files